Sources/Dependencies/DependencyValues.swift

```diff
@@ -461,3 +477,25 @@ public final class CachedValues: @unchecked Sendable {
     }
   }
 }
+
+struct TypeIdentifier: Hashable {
+  let id: ObjectIdentifier
+  #if DEBUG
+  let typeName: String
+  #endif
+
+  init<T>(_ type: T.Type) {
+    self.id = ObjectIdentifier(type)
+    #if DEBUG
+    self.typeName = Dependencies.typeName(type)
+#endif
+  }
```

Comment:
@mbrandonw Alternately, if we don't want to eagerly do formatting:

```suggestion
struct TypeIdentifier: Hashable {
  let id: ObjectIdentifier
  #if DEBUG
    let base: Any.Type
  #endif

  init<T>(_ type: T.Type) {
    self.id = ObjectIdentifier(type)
    #if DEBUG
      self.base = type
    #endif
  }
```